sync/class.jetpack-sync-actions.php

```diff
@@ -178,18 +182,68 @@ static function send_data( $data, $codec_name, $sent_timestamp, $queue_id ) {
 	}
 
 	static function do_initial_sync( $new_version = null, $old_version = null ) {
-		$initial_sync_config = array(
+		$initial_sync_config = self::get_update_full_sync_config();
+
+		if ( $old_version && ( version_compare( $old_version, '4.2', '<' ) ) ) {
+			$initial_sync_config['users'] = 'initial';
+		}
+
+		if ( is_multisite()  ) {
+			// stagger initial syncs for multisite blogs so they don't all pile on top of each other
+			if ( is_main_site() ) {
+				self::schedule_full_sync_multisite();
+			}
+		} else {
+			self::do_full_sync( $initial_sync_config );
+		}
+	}
+
+	static function get_update_full_sync_config() {
+		return array(
 			'options' => true,
 			'network_options' => true,
 			'functions' => true,
 			'constants' => true,
 		);
+	}
 
-		if ( $old_version && ( version_compare( $old_version, '4.2', '<' ) ) ) {
-			$initial_sync_config['users'] = 'initial';
+	static function schedule_full_sync_multisite(  ) {
+		wp_schedule_single_event( time() , 'jetpack_sync_update_multisite_cron' );
+	}
+	
+	static function update_multisite_full_sync( $offset = 0 ) {
+		$batch = 20;
+
+		// doesn't exist pre 4.6
+		if ( function_exists( 'get_sites' ) ) {
+			$args     = array(
+				'network_id' => get_current_network_id(),
+				'spam'       => 0,
+				'deleted'    => 0,
+				'archived'   => 0,
+				'number'     => $batch,
+				'offset'     => $offset,
+				'fields'     => 'ids',
+				'order'      => 'DESC',
+				'orderby'    => 'id',
+			);
+			$site_ids = get_sites( $args );
+		} else  {
+			global $wpdb;
+			$site_ids = $wpdb->get_results( "SELECT blog_id FROM {$wpdb->blogs} WHERE site_id = '{$wpdb->siteid}' AND spam = '0' AND deleted = '0' AND archived = '0' ORDER BY registered DESC LIMIT {$offset}, {$batch}", ARRAY_A );
+		}
+		
+		foreach ( (array) $site_ids as $site_id ) {
+			switch_to_blog( $site_id );
+			if ( self::sync_allowed() ) {
```

Comment:
do_full_sync already calls `self::sync_allowed()`, so you don't need this check.